src/enable_variable_ops/ngraph_replace_op_utilities.cc

```diff
@@ -216,6 +216,44 @@ Status ReplaceOutputEdges(Graph* graph, Node* node, Node* replacement) {
   return Status::OK();
 }
 
+bool IsValidateShape(Node* node) {
+  bool validate_shape;
+  GetNodeAttr(node->attrs(), "validate_shape", &validate_shape);
+  return validate_shape;
+}
+
+Status StoreRefTypeOutputs(Node* node, std::set<Node*>* ref_list) {
+  for (auto edge : node->out_edges()) {
+    Node* dst = edge->dst();
+    if (IsRefType(dst->input_type(edge->dst_input()))) {
+      NGRAPH_VLOG(4) << "Found a ref type output " << dst->name();
+      // Check if the node is Assign node and it's attribute variable_shape
+      if (dst->type_string() == "Assign" && !IsValidateShape(dst)) {
+        NGRAPH_VLOG(4) << "The attribute validate_shape for Assign is false ";
+        // If the dst node is Assign and attr: variable_shape is false then
+        // we do not want to capture the Assign or any of the ref nodes leading
+        // from it or the variable node or ref nodes that lead to it.
+        // So clear the ref list and return.
+        ref_list->clear();
+        return Status::OK();
+      } else {
+        // In all other cases:
+        // 1. Assign, variable_shape = true
+        // 2. AssignAdd
+        // 3. AssignSub
+        // 4. ApplyGradientDescent
+        // which are all ref types, add to the ref list if not already added
```

Comment:
It seems all the outputs that are of RefType are added and not just the 4 ops that we capture

Reply:
We need a check if its one of the 4 types mentioned above. This (List of assign type ops that ng-bridge supports) should be a function/table in utils or somewhere, so that we can keep adding to it.

Reply:
Yes, we would add all the outputs that are of RefType but only the 4 ops will be captured that are present in the CAPTURE_REPLACE_OP_MAP, but I could also add a check to add only those 4 to the ref list.

Reply:
I see, so its functionally correct now i guess.

Also `nodes_to_capture` is actually `possible_nodes_to_capture`, right? (because its a necessary but not sufficient condition for a node to be captured is that it ends up in `nodes_to_capture`/`possible_nodes_to_capture`)

Reply:
The check for being present in CAPTURE_REPLACE_OP_MAP is missing in the main function also